src/components/views/rooms/MessageComposerInput.js

```diff
@@ -477,19 +479,7 @@ export default class MessageComposerInput extends React.Component {
     }
 
     sendTyping(isTyping) {
-        if (!SettingsStore.getValue('sendTypingNotifications')) return;
-        if (SettingsStore.getValue('lowBandwidth')) return;
-        MatrixClientPeg.get().sendTyping(
-            this.props.room.roomId,
-            this.isTyping, TYPING_SERVER_TIMEOUT,
-        ).done();
-    }
-
-    refreshTyping() {
-        if (this.typingTimeout) {
-            clearTimeout(this.typingTimeout);
-            this.typingTimeout = null;
-        }
+        TypingStore.sharedInstance().setSelfTyping(this.props.room.roomId, isTyping);
```

Comment:
nit: shall we just replace the calls to `sendTyping` with this line?

Reply:
it's a bit long, and the utility of having a function on the composer to hammer (which in turn hammers the TypingStore) doesn't seem like the worst idea

Reply:
don't have a strong opinion, feel free to do what you think is best.